onnxruntime/core/framework/session_state.h

```diff
@@ -187,15 +199,14 @@ class SessionState {
   void SetDataTransferMgr(const DataTransferManager* data_transfer_mgr) { data_transfer_mgr_ = data_transfer_mgr; }
 
   std::vector<BufferUniquePtr>& GetMutableWeightsBuffers() { return weights_buffers_; }
-  void CalculateNodeIndexInfo();
   const NodeIndexInfo& GetNodeIndexInfo() const;
 
  private:
   ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE(SessionState);
 
   // cache of the constructed kernels to avoid spending construction
   // time per executor
-  std::unordered_map<NodeIndex, std::unique_ptr<OpKernel>> session_kernels_;
+  std::vector<OpKernel*> session_kernels_;
```

Comment:
This line is the major change.

Reply:
Why are we not using unique_ptr so the ownership semantics are clear?

---
In reply to: [316516508](https://github.com/microsoft/onnxruntime/pull/1672#discussion_r316516508) [](ancestors = 316516508)